### PR TITLE
Replace strstr with StringPtr::contains

### PR DESCRIPTION
**Pull request by justin-mp**
This series of commits replaces uses of strstr with StringPtr::contains.

There's still 1 place where workerd uses strstr, but that's doing index arithmetic and I'll fix it in a follow-up PR.

### CODE REVIEW

File: src/workerd/jsg/exception.c++ [hunk 1]
```diff
@@ -153,7 +153,7 @@ bool isTunneledException(kj::StringPtr internalMessage) {
 }
 
 bool isDoNotLogException(kj::StringPtr internalMessage) {
-  return strstr(internalMessage.cStr(), "worker_do_not_log") != nullptr;
+  return internalMessage.contains( "worker_do_not_log");
```
Comment (ohodson) [inline src/workerd/jsg/exception.c++:156]:
```suggestion
  return internalMessage.contains("worker_do_not_log");
```

Comment (justin-mp) [reply src/workerd/jsg/exception.c++]: Fixed!

File: src/workerd/api/trace.c++ [hunk 2]
```diff
@@ -276,11 +276,11 @@ jsg::Dict<jsg::ByteString, jsg::ByteString> TraceItem::FetchEventInfo::Request::
       //(name == "authorization"_kj) || // covered below
       (name == "cookie"_kj) ||
       (name == "set-cookie"_kj) ||
-      (strstr(name.cStr(), "auth") != nullptr) ||
-      (strstr(name.cStr(), "jwt") != nullptr) ||
-      (strstr(name.cStr(), "key") != nullptr) ||
-      (strstr(name.cStr(), "secret") != nullptr) ||
-      (strstr(name.cStr(), "token") != nullptr)
+      name.contains("auth") ||
```
Comment (ohodson) [inline src/workerd/api/trace.c++:279]: Would suffixing these string literals with `_kjc` avoid runtime calls to `strlen` going from a C-string to a `kj::StringPtr`?

**Review by ohodson — APPROVED** [review]

File: src/workerd/api/trace.c++ (hunk 2 above)
Comment (justin-mp) [reply src/workerd/api/trace.c++]: Done.

Comment (vickykont) [reply src/workerd/api/trace.c++]: Nit; looks like there's one more of these in `exception.c++`.

**Review by vickykont — APPROVED** [review]
Leaving just a small nit.

File: src/workerd/api/trace.c++ (hunk 2 above)
Comment (justin-mp) [reply src/workerd/api/trace.c++]: `exception.c++` fixed.

Comment (vickykont) [reply src/workerd/api/trace.c++]: Thanks!

Comment (kentonv) [reply src/workerd/api/trace.c++]: FWIW `_kjc` produces a `kj::ConstString` whereas `_kj` produces `kj::StringPtr`. The latter would have worked here and feels slightly lighter although probably makes no real difference.

Comment (ohodson) [reply src/workerd/api/trace.c++]: Thanks. I posed the question as I didn't see such an initializer in string.h, but good to know.

Comment (kentonv) [reply src/workerd/api/trace.c++]: @ohodson Hmm, I may have misspoke... but the other thing is, I'd expect the compiler can elide `strlen()` when the parameter is known at compile time.

**Pull request merged**